cob_light/common/include/turnIndicatorMode.h

```diff
@@ -14,79 +14,70 @@
  * limitations under the License.
  */
 
-
-
 #ifndef TURNINDICATORMODE_H
 #define TURNINDICATORMODE_H
 
 #include <mode.h>
 
-class TurnIndicatorMode : public Mode
-{
+class TurnIndicatorMode : public Mode {
 public:
-    TurnIndicatorMode(color::rgba color, size_t num_leds, int dir, int priority = 0, double freq = 0.25, int pulses = 0, double timeout = 0)
-        :Mode(priority, freq, pulses, timeout), _num_leds(num_leds)
-    {
-      _inc = (1. / UPDATE_RATE_HZ) * _freq;
+  TurnIndicatorMode(color::rgba color, size_t num_leds, int dir,
+                    int priority = 0, double freq = 0.25, int pulses = 0,
+                    double timeout = 0)
+      : Mode(priority, freq, pulses, timeout), _num_leds(num_leds),
+        _timer_inc(0), _toggle(false) {
+    _color = color;
+    _inc = (1. / UPDATE_RATE_HZ) * _freq;
 
-      _color = color;
-      _colors.assign(_num_leds, color::rgba());
-      _colors[0] = color;
-      _pos = 0;
-      _dir = dir;
-      if(_num_leds%2 == 0)
-      {
-        _center_pos = _num_leds/2 - 1;
-        if(_dir < 0)
-        _center_pos--;
+    _colors.assign(_num_leds, color::rgba());
+    _dir = dir;
+    if (dir == -1) { // left
```

Comment:
left turn is `dir == 1`

Reply:
dir == -1 for left is defined in the mode factory, led offset should be adjusted as it seems:
- https://github.com/ipa320/cob_driver/blob/kinetic_dev/cob_light/common/src/modeFactory.cpp#L161-L168
```
    case cob_light::LightModes::TURN_LEFT:
        mode.reset(new TurnIndicatorMode(color, colorO->getNumLeds(), -1, requestMode.priority, requestMode.frequency,
                                      requestMode.pulses, requestMode.timeout));
        break;

    case cob_light::LightModes::TURN_RIGHT:
        mode.reset(new TurnIndicatorMode(color, colorO->getNumLeds(), 1, requestMode.priority, requestMode.frequency,
                                      requestMode.pulses, requestMode.timeout));
```